### ISSUE TITLE
[Follow up to #28904] Clearing browsing data for a specific site doesn't work

### ISSUE THREAD

**Issue by hffvld** [brave/brave-browser#43023]
### Description

Follow-up to #28904.

### Steps to reproduce

1. Launch Brave
2. Open `https://dev-pages.bravesoftware.com/fingerprinting/farbling.html ` and `https://dev-pages.brave.software/fingerprinting/farbling.html ` in 2 different tabs
3. Generate fingerprints at the test page > *Note the `WebGL Drawing` hash value (`Hash A`)*
4. Go to the `Settings` > Site settings > All sites
5. Clear browsing data for a test site
6. Reload the test page and generate fingerprints again > *Observe that the new hash value (`Hash B`) is the same as `Hash A`*

### Actual result

Clearing browsing data for a specific site doesn't work. And the new hash value (`Hash B`) is the same as `Hash A` after clearing specific site data.

------------------------

|Test page 1 - before clearing data|Test page 2 - before clearing data|
|:-:|:-:|
| ![1](https://github.com/user-attachments/assets/0eba9293-a61e-4645-870b-85a2993befc8) | ![2](https://github.com/user-attachments/assets/50c94c04-8de8-4100-b8e6-add94fbc5989) |
| ![1](https://github.com/user-attachments/assets/06f153c9-8a06-4078-b17c-6611edf77c87) | ![2](https://github.com/user-attachments/assets/1ba600ca-603c-4cb4-b327-88a93195c6ff) |
|Test page 1 - after clearing data|Test page 2 - after clearing data|
| ![1](https://github.com/user-attachments/assets/dfc93f63-c0f3-4c90-9391-f60ee00461f4) | ![2](https://github.com/user-attachments/assets/7aa4bd86-b2ce-43ab-8d69-4ad8e512a811) |

### Expected result

After clearing specific site data, the new hash value (`Hash B`) must be different from `Hash A`.

------------------

`Desktop behavior`

|Test page 1 - before clearing data|Test page 2 - before clearing data|
|:-:|:-:|
| ![1](https://github.com/user-attachments/assets/90d53bd4-c38a-4252-bb26-3643a50bd9c5) | ![2](https://github.com/user-attachments/assets/6332eccb-0a9b-4f08-9aec-44675b553102) |
| ![1](https://github.com/user-attachments/assets/796efc8f-3ee8-41c2-9e16-1bce7e7fff63) | ![2](https://github.com/user-attachments/assets/ac3722aa-2d5b-43fe-9f9d-3d20e654d883) |
|Test page 1 - after clearing data|Test page 2 - after clearing data|
| ![1](https://github.com/user-attachments/assets/bf549716-993a-4eff-8240-ff33d3eabc31) | ![2](https://github.com/user-attachments/assets/24c553b1-2c91-4eba-abaf-8f9439c8bbab) |

### Reproduces how often

Easily reproduced

### Brave version

Brave build: 1.75.105
Chromium: 132.0.6834.46 (Official Build) canary (64-bit) 

### Device

- Brand/model: Galaxy Z Fold 6
- Android version: 14

### Channel information

- [ ] release (stable)
- [ ] beta
- [x] nightly

### Reproducibility

- [ ] with Brave Shields disabled
- [ ] with Brave Rewards disabled
- [ ] in the latest version of Chrome

**Comment by MadhaviSeelam** [comment]
@hffvld looks like it's only a mobile issue as desktop is working as expected?  I could not reproduce in Win 11 x64

**Comment by hffvld** [comment]
Yes @MadhaviSeelam, I couldn't reproduce this issue on Desktop (macOS) as well.

**Comment by hffvld** [comment]
Verified on `Pixel 6` using version(s):
```
Device/OS: Pixel 6 / oriole-user 15 AP4A.241205.013 release-keys
Brave build: 1.76.52 
Chromium: 133.0.6943.54 (Official Build) beta (64-bit) 
```
---------------------------

**STEPS:**
1. Follow the STR/TP from https://github.com/brave/brave-browser/issues/43023#issue-2753697408
2. *Verify*

**ACTUAL RESULTS:**
- Verified that the new hash value (Hash B) differs from Hash A after clearing site data.

---------------------------

| 1 | 2 |
|:-:|:-:|
| ![1](https://github.com/user-attachments/assets/bf066087-b330-4cec-9246-fcede17b49f2) | ![2](https://github.com/user-attachments/assets/52e84bca-0ce4-4636-bb8a-a19131e565f0) |
| ![1](https://github.com/user-attachments/assets/77b03346-0efa-4c58-912c-c10c1be862df) | ![2](https://github.com/user-attachments/assets/e005f5b2-9572-419b-b976-051cafa220aa) |
| ![1](https://github.com/user-attachments/assets/f7efec39-5bab-4d15-a7f3-2d9dc7777abc) | ![2](https://github.com/user-attachments/assets/e1251dc3-152e-4486-b7c3-6ad4b5cff1a2) |

**Comment by hffvld** [comment]
Verified on `Galaxy Tab S8` using version(s):
```
Device/OS: Galaxy Tab S8 / gts8wifixx-user 14 UP1A.231005.007 release-keys
Brave build: 1.76.58 
Chromium: 133.0.6943.98 (Official Build) beta (64-bit) 
```
---------------------------

**STEPS:**
1. Follow the STR/TP from https://github.com/brave/brave-browser/issues/43023#issue-2753697408
2. *Verify*

**ACTUAL RESULTS:**
- Verified that the new hash value (Hash B) differs from Hash A after clearing site data.

---------------------------

| 1 | 2 |
|:-:|:-:|
| ![1](https://github.com/user-attachments/assets/42474162-8df6-4f0d-9ff4-7243ddd0761f) | ![2](https://github.com/user-attachments/assets/a6f9ca0c-837c-4405-a589-c0bbace8621d) |
| ![1](https://github.com/user-attachments/assets/ef4257e7-4436-48a0-b6d0-9ff7eadfde17) | ![2](https://github.com/user-attachments/assets/2dd74a00-0f3b-4122-a1d3-8c348276216f) |
| ![1](https://github.com/user-attachments/assets/a9395e5f-799b-474b-bd61-21109df33ffd) | ![2](https://github.com/user-attachments/assets/358c4142-2888-484f-b1b2-3ff4b188c958) |
